src/main/java/org/candlepin/model/ContentQueryBuilder.java

```diff
@@ -0,0 +1,666 @@
+     *
+     * @param queryChunks
+     *  the list of query chunks to receive the query components for filtering on custom contents
+     *
+     * @param criteriaChunks
+     *  the list of criteria chunks to receive the predicates for filtering on custom contents
+     *
+     * @param parameters
+     *  the map of parameters to receive the specific query arguments provided to the builder
+     */
+    private void buildCustomContentsComponents(List<String> queryChunks, List<String> criteriaChunks,
+        Map<String, Object> queryArgs) {
+
+        List<String> namespaces = this.owners.values()
+            .stream()
+            .map(Owner::getKey)
+            .toList();
+
+        switch (this.custom) {
+            case EXCLUSIVE:
+                if (!namespaces.isEmpty()) {
+                    criteriaChunks.add("content.namespace IN (:org_namespaces)");
+                    queryArgs.put("org_namespaces", namespaces);
+                }
+                else {
+                    criteriaChunks.add("content.namespace != ''");
+                }
+                break;
+
+            case EXCLUDE:
+                criteriaChunks.add("content.namespace = ''");
+                break;
+
+            default:
+            case INCLUDE:
+                if (!namespaces.isEmpty()) {
+                    criteriaChunks.add("(content.namespace = '' OR content.namespace IN (:org_namespaces))");
+                    queryArgs.put("org_namespaces", namespaces);
+                }
+        }
+    }
+
+    /**
+     * Builds the components of the query that make up the content ID filter.
+     *
+     * @param queryChunks
+     *  the list of query chunks to receive the query components for filtering on content IDs
+     *
+     * @param criteriaChunks
+     *  the list of criteria chunks to receive the predicates for filtering on content IDs
+     *
+     * @param parameters
+     *  the map of parameters to receive the specific query arguments provided to the builder
+     */
+    private void buildContentIdFilterComponents(List<String> queryChunks, List<String> criteriaChunks,
+        Map<String, Object> queryArgs) {
+
+        if (!this.contentIds.isEmpty()) {
+            criteriaChunks.add("content.content_id IN (:content_ids)");
+            queryArgs.put("content_ids", this.contentIds);
+        }
+    }
+
+    /**
+     * Builds the components of the query that make up the content name filter.
+     *
+     * @param queryChunks
+     *  the list of query chunks to receive the query components for filtering on content names
+     *
+     * @param criteriaChunks
+     *  the list of criteria chunks to receive the predicates for filtering on content names
+     *
+     * @param parameters
+     *  the map of parameters to receive the specific query arguments provided to the builder
+     */
+    private void buildContentNameFilterComponents(List<String> queryChunks, List<String> criteriaChunks,
+        Map<String, Object> queryArgs) {
```

Comment:
Update to content labels instead of content name

Reply:
Fixed